api/src/ddoc-extraction.js

```diff
@@ -43,17 +44,17 @@ const areAttachmentsEqual = (oldDdoc, newDdoc) => {
 };
 
 const extractCompiledDdoc = (newDdoc, deploy_info) => {
+  // update the deploy info in the medic-client ddoc
+  if (newDdoc._id === CLIENT_DDOC_ID && (deploy_info || newDdoc.deploy_info)) {
+    newDdoc.deploy_info = deploy_info;
+  }
+
```

Comment:
Theoretically, `medic-client` always exists at this point in prod (cause Horti warms the views for all ddocs and then renames them), but might not for dev deployments and there's an e2e test lower that requires it.